pandas/tests/arithmetic/test_numeric.py

```diff
@@ -1317,3 +1317,13 @@ def test_dataframe_div_silenced():
     )
     with tm.assert_produces_warning(None):
         pdf1.div(pdf2, fill_value=0)
+
+
+def test_inplace_arithmetic_operation_on_series_updating_parent_dataframe():
+    # https://github.com/pandas-dev/pandas/issues/36373
+    df = pd.DataFrame({"A": [1, 2, 3]})
+    s = df["A"]
+    s += 1
+    assert s is df["A"]
```

Comment:
can we also `assert df['A'].to_numpy().base is s.to_numpy().base`

Reply:
I guess if we do this we no longer need the cacher machinery. At first glance his appears a big change so will open a separate PR as an alternative to this one.